### DISCUSSION TITLE
Security logging of logged in users / failed logons - delayed login / brute force protection / 2FA support

### DISCUSSION THREAD

**Discussion by MsPoppins** [wavelog/wavelog#659] [Ideas]
Hello @all,
I've installed wavelog on shared webhosting.
Works very nice, thank you!

Maybe I overlooked it, I miss some security features:
- are logons logged somewhere?
- are failed logons logged, if wrong login is detected delay the logon window
- brute force protection

Are there any plans to support 2FA?

73, MsPoppins

**Comment by raspi-coder** [comment]
I second the mentioned topics. 

Having some sort of file based logging would be helpful. This way fail2ban could do its job - at least on a dedicated (virtual) server where you have access to the filesystem. But 2FA and brute force protection would also work on shared systems and enhance security significantly.

**Comment by PE1PQX** [comment]
I am also in favour of 2FA login, for example with a Yubi-key...?

**Comment by int2001** [comment]
jumping into this discussion.
Adding 2FA wouldn't mean "only add 2FA".
it means, that the API-Access has to be completely refactored. As long as one is able to modify/upload/read QSOs and things via API the 2FA would be senseless.

e.g.: Clublog has 2FA. The 2FA is absolutely nonsense there, because you're able to bypass it via the API. 
And tbh: We're not friends of snakeoil-security

**Comment by HB9HIL** [comment]
Regarding 2FA: We are working on #2978 which allows you to use thirdparty idp's for 2fa as you like :)
